x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/querydsl/agg/GroupByScriptKey.java

```diff
@@ -39,8 +39,12 @@ public TermsValuesSourceBuilder asValueSource() {
                 .order(direction().asOrder())
                 .missingBucket(true);
 
-        if (script.outputType().isNumeric()) {
-            builder.valueType(ValueType.NUMBER);
+        if (script.outputType().isInteger) {
+            builder.valueType(ValueType.LONG);
+        } else if (script().outputType().isRational) {
```

Comment:
Why here accessing the `script` via the public method, and for the other values checks directly the field?

Reply:
no reason :-), will change, thx!